wimm.Secundatives/Range.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+
+namespace wimm.Secundatives
+{
+    /// <summary>
+    /// A range for an <see cref="IComparable{T}"/> value.
+    /// </summary>
+    public class Range<T> where T : IComparable<T>
+    {
+        /// <summary>
+        /// The minimum value.
+        /// </summary>
+        public T Min { get; }
+
+        /// <summary>
+        /// The maximum value.
+        /// </summary>
+        public T Max { get; }
+
+        /// <summary>
+        /// Indicates whether <see cref="Min"/> and <see cref="Max"/> are excluded from the range.
+        /// When <c>false</c>, values be greater than or equal to <see cref="Min"/> and less than
+        /// or equal to <see cref="Max"/> are considered in range. When <c>true</c> values must be
+        /// strictly greater than <see cref="Min"/> and less than <see cref="Max"/>.
+        /// </summary>
+        public bool Exclusive { get; }
+
+        /// <summary>
+        /// Initializes a new, non-exclusive <see cref="Range{T}"/>.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="min"/> or <paramref name="max"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="max"/> is less than <paramref name="min"/>.
+        /// </exception>
+        public Range(T min, T max) : this(min, max, false) { }
```

Comment:
I think that this being true by default would be the more common scenario that comes up since a lot of range implementations in diff langs behave that way.